kliff/models/model.py

```diff
@@ -335,178 +230,155 @@ def echo_model_params(
-            Initial guess of the parameter component. If `DEFAULT` (case insensitive),
-            the value from the calculator is used as the initial guess.
-
-        The 2nd and 3rd elements are optional.
-
-        If 2 elements are provided:
-
-        2nd element: `FIX` (case insensitive)
-            If `FIX`, the corresponding component will not be optimized.
-
-        If 3 elements are provided:
-
-        2nd element: float or `INF` (case insensitive)
-            Lower bound of the parameter. `INF` indicates that the lower bound is
-            negative infinite, i.e. no lower bound is applied.
-
-        3rd element: float or `INF` (case insensitive)
-            Upper bound of the parameter. `INF` indicates that the upper bound is
-            positive infinite, i.e. no upper bound is applied.
-
-        Instead of reading fitting parameters from a file, you can also setting them
-        using a dictionary by calling the `set_opt_params()` or `set_one_opt_params()`
-        method.
-
-        Args:
-            filename: path to file that includes the fitting parameters.
-
-        Example:
-
-            # put the below in a file, say `model_params.txt` and you can read the fitting
-            # parameters by this_class.read(filename="model_params.txt")
-
-            A
-            DEFAULT
-            1.1
-
-            B
-            DEFAULT FIX
-            1.1     FIX
-
-            C
-            DEFAULT  0.1  INF
-            1.0      INF  2.1
-            2.0      FIX
-        """
-        self.opt_params.read(filename)
-        self.model_params = self._inverse_transform_params(self.opt_params.model_params)
+        pass
 
-        # reset influence distance in case it depends on parameters and changes
-        self.init_influence_distance()
+    def set_params_mutable(self, opt_params: List[str]):
+        pass
 
     def set_opt_params(self, **kwargs):
-        """
-        Set the parameters that will be optimized.
-
-        One or more parameters can be set. Each argument is for one parameter, where the
-        argument name is the parameter name, the value of the argument is the
-        settings(including initial value, fix flag, lower bound, and upper bound).
-
-        The value of the argument should be a list of list, where each inner list is for
-        one component of the parameter, which can contain 1, 2, or 3 elements.
-         See `~kliff.model.model.Model.read_opt_params()` for the options of the elements.
-
-        Example:
-           instance.set(A=[['DEFAULT'], [2.0, 1.0, 3.0]], B=[[1.0, 'FIX'], [2.0, 'INF', 3.0]])
-        """
-        self.opt_params.set(**kwargs)
-        self.model_params = self._inverse_transform_params(self.opt_params.model_params)
+        keys = list(kwargs.keys())
+        optimizable_keys = []
+        for key in keys:
+            if len(kwargs[key][0]) == 2:
+                try:
+                    if kwargs[key][0][1] == "fix":
+                        continue
```

Comment:
Similar for a comments in `kim.py`. `fix` means a value needs not need to be optimized. We need to update to code to reflect this.

Reply:
Please see above unresolved conversation. I am marking this one as resolved for sanitization.